api/src/analysis.rs

```diff
@@ -186,6 +186,7 @@ async fn analyze_package_inner(
       .analyzer
       .get_parsed_source(module.specifier())
     {
+      check_for_banned_extensions(&parsed_source)?;
```

Comment:
actually at this point we would've already run into a `graph` error.
still thought it might be a good idea to have a final sanity check.